bindings/rust/src/bindings/mod.rs

```diff
@@ -194,6 +198,64 @@ impl Drop for KZGSettings {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct Sample(Vec<Bytes32>);
+
+impl Sample {
+    pub fn as_slice(&self) -> &[Bytes32] {
+        self.0.as_slice()
+    }
+
+    pub fn from_field_elements(
+        elements: &[Bytes32],
+        kzg_settings: &KZGSettings,
+    ) -> Result<Self, Error> {
+        if elements.len() != kzg_settings.sample_size as usize {
+            return Err(Error::InvalidFieldElementsLength);
+        }
+        Ok(Self(Vec::from(elements)))
+    }
+}
+
+pub struct SampleMatrix<const ROWS: usize, const COLS: usize> {
+    matrix: Vec<Vec<Sample>>,
+}
```

Comment:
We've kind of discussed this, but I think it would be better not to use constant generics.